pkg/asset/ignition/bootstrap/baremetal/template.go

```diff
@@ -111,14 +117,47 @@ func GetTemplateData(config *baremetal.Platform, networks []types.MachineNetwork
 	templateData.ExternalStaticDNS = config.BootstrapExternalStaticDNS
 	templateData.ExternalMACAddress = config.ExternalMACAddress
 
-	_, externalURLv6 := externalURLs(config.APIVIPs)
+	// If the user has manually set disableVirtualMediaTLS to False in the Provisioning CR,
+	// then enable TLS in ironic
+	protocol := "http"
+	provisioningCR := struct {
+		apiVersion string `yaml:"apiVersion"`
+		kind       string `yaml:"kind"`
+		metadata   struct {
+			name string `yaml:"name"`
+		} `yaml:"metadata"`
+		spec struct {
+			provisioningInterface     string `yaml:"provisioningInterface"`
+			provisioningIP            string `yaml:"provisioningIP"`
+			provisioningNetworkCIDR   string `yaml:"provisioningNetworkCIDR"`
+			provisioningNetwork       string `yaml:"provisioningNetwork"`
+			provisioningDHCPRange     string `yaml:"provisioningDHCPRange"`
+			provisioningOSDownloadURL string `yaml:"provisioningOSDownloadURL"`
+			watchAllNamespaces        bool   `yaml:"watchAllNamespaces"`
+			disableVirtualMediaTLS    bool   `yaml:"disableVirtualMediaTLS"`
```

Comment:
This has to be a pointer to boolean, otherwise you don't distinguish "not present" from "is set to false".

Reply:
Done